src/components/ArticleList.jsx

```diff
@@ -2,21 +2,44 @@ import { useEffect, useState } from "react";
 import { fetchArticles } from "../../newsApi";
 import ArticleCard from "./ArticleCard";
 import Pagination from "./Pagination";
-import { useParams } from "react-router-dom";
+import { useParams, useSearchParams } from "react-router-dom";
+import Options from "./Options";
 
 const ArticleList = () => {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
   const [currentPage, setCurrentPage] = useState(1);
   const [articlesPerPage, SetArticlesPerPage] = useState(10);
+  const [sortBy, setSortBy] = useState("created_at");
+  const [orderBy, setOrderBy] = useState("ASC");
   const { topic } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sortArticles = (articles, orderByQuery) => {
+    if (orderByQuery === "DESC" && sortBy === "comment_count") {
+      return articles.sort(
+        (a, b) => Number(b.comment_count) - Number(a.comment_count)
+      );
+    } else if (orderByQuery === "ASC" && sortBy === "comment_count") {
+      return articles.sort(
+        (a, b) => Number(a.comment_count) - Number(b.comment_count)
+      );
+    }
+    return articles;
```

Comment:
A lot of sorting going on in the front end. If you got to the sorting part in your backend, you can get the API to do the work instead 😄